doc/api/n-api.md

```diff
@@ -3705,6 +3706,93 @@ NAPI_EXTERN napi_status napi_get_uv_event_loop(napi_env env,
 - `[in] env`: The environment that the API is invoked under.
 - `[out] loop`: The current libuv loop instance.
 
+<!-- it's very convenient to have all the anchors indexed -->
+<!--lint disable no-unused-definitions remark-lint-->
+## Asynchronous Thread-safe Function Calls
+JavaScript functions can normally only be called from a native addon's main
+thread. If an addon creates additional threads then N-API functions that require
+a `napi_env`, `napi_value`, or `napi_ref` must not be called from those threads.
+
+This API provides the type `napi_async_function` as well as APIs to create,
+destroy, and call objects of this type. `napi_async_function` creates a
+permanent reference to a `napi_value` that holds a JavaScript function, and
+uses `uv_async_t` from libuv to coordinate calls to the JavaScript function from
+all threads.
+
+Since `uv_async_t` is used in the implementation, the caveat whereby multiple
+invocations on the secondary thread may result in only one invocation of the
+JavaScript function also applies to `napi_async_function`.
+
+### napi_create_async_function
+<!-- YAML
+added: UPDATEME
+-->
+```C
+NAPI_EXTERN napi_status
+napi_create_async_function(napi_env env,
+                           napi_value func,
+                           void* data,
+                           void* context,
+                           size_t argc,
+                           napi_async_function_marshal marshal_cb,
+                           napi_async_function_process_result process_result_cb,
+                           napi_async_function* result);
+```
+
+- `[in] env`: The environment that the API is invoked under.
+- `[in] func`: The JavaScript function to call from another thread.
+- `[in] data`: Optional data to attach to the resulting `napi_async_function`.
+- `[in] context`: Optional context associated with `data`.
+- `[in] argc`: Number of arguments the JavaScript function will have.
+- `[in] marshal_cb`: Optional callback to convert `data` and `context` to
+JavaScript function arguments. The callback will always be called on the main
+thread.
+- `[in] process_result_cb`: Optional callback to handle the return value and/or
+exception resulting from the invocation of the JavaScript function. The callback
+will always be called on the main thread.
+- `[out] result`: The asynchronous thread-safe JavaScript function.
+
+### napi_call_async_function
+<!-- YAML
+added: UPDATEME
```

Comment:
`REPLACEME` is picked up by tooling automatically :) I think you should also update the `NAPI_VERSION` #define in node_api.cc?

Reply:
Aaaah! Forgot the magic word :)